control-plane-operator/controllers/hostedcontrolplane/oauth/route.go

```diff
@@ -9,6 +9,12 @@ import (
 	"github.com/openshift/hypershift/support/util"
 )
 
+const (
+	// hyperShiftOAuthRouteLabel is a label added to OAuth routes so that they
+	// can be selected via label selector in a dedicated router shard.
+	hyperShiftOAuthRouteLabel = "hypershift.openshift.io/oauth"
```

Comment:
have we considered this vs semantic which does is not tied to the purpose of the route, e.g `hypershift.openshift.io/discoverable`?

Reply:
hmm, a generic label does make sense if in the future we have another route that we want to expose publicly. But maybe something more explicit like `hypershift.openshift.io/public-route` ?